swe4103/swe4103-project
# Feat scrum 41

## 🛠️ What Type of PR is This?


## 📄 Description

Backend Changes:
Updated the authController to handle change password requests.
Enhanced the usersService with a method to change user passwords, including validation of the current password.
Added changePasswordSchema for input validation during password change requests.
Frontend Changes:
Added the ChangePasswordForm component to allow users to submit password changes.
Updated App.jsx to include the new ChangePasswordForm view.
Dependencies and Configurations:
Modified project dependencies as needed.

## 🔗 Jira Ticket

[SCRUM-41](https://swe4103-pm-board.atlassian.net/browse/SCRUM-41)

## 🧾 Test Evidence

Frontend:
Tests added for the new ChangePasswordForm component.
Run tests via yarn test to confirm that the form is validated correctly
Steps to test:
Render the ChangePasswordForm component in the application.
Enter mismatched passwords to verify that validation errors are shown.
Enter matching passwords and verify the successful submission.

Backend:
authController:
Tests for schema validation, user retrieval, password comparison, and successful password change.
users service:
Tests for changing passwords with correct and incorrect current passwords and user-not-found scenarios.

Note: due to the lack of a valid database endpoint, I was unable to complete the backend tests. However, the test files are included in the API test module.

## 🧪 Added/Updated tests?
